fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MaterializationContext.java

```diff
@@ -15,50 +15,67 @@
 
 package com.starrocks.sql.optimizer;
 
+import com.google.common.collect.Lists;
 import com.starrocks.catalog.MaterializedView;
 import com.starrocks.catalog.Table;
 import com.starrocks.sql.optimizer.base.ColumnRefFactory;
 import com.starrocks.sql.optimizer.operator.logical.LogicalOlapScanOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
 
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 public class MaterializationContext {
-    private MaterializedView mv;
+    private final MaterializedView mv;
     // scan materialized view operator
     private LogicalOlapScanOperator scanMvOperator;
     // logical OptExpression for query of materialized view
-    private OptExpression mvExpression;
+    private final OptExpression mvExpression;
 
-    private ColumnRefFactory mvColumnRefFactory;
+    private final ColumnRefFactory mvColumnRefFactory;
 
-    // logical OptExpression for query
-    private OptExpression queryExpression;
+    private final ColumnRefFactory queryRefFactory;
 
-    private ColumnRefFactory queryRefFactory;
-
-    private OptimizerContext optimizerContext;
+    private final OptimizerContext optimizerContext;
 
     private Map<ColumnRefOperator, ColumnRefOperator> outputMapping;
 
-    private Set<String> mvPartitionNamesToRefresh;
+    private final Set<String> mvPartitionNamesToRefresh;
+
+    private final List<Table> baseTables;
+
+    private final Set<ColumnRefOperator> originQueryColumns;
 
-    private List<Table> baseTables;
+    // tables both in query and mv
+    private final List<Table> intersectingTables;
 
-    public MaterializationContext(MaterializedView mv,
+    private List<List<Table>> matchedTableLists;
```

Comment:
List<Set<Table>> matchedTableLists? and plz add some  comment for this

Reply:
use groupid to replace List<Table>